### DISCUSSION TITLE
How to pass Chinese information in routing parameters? I tried to operate, but did not find a matching route

### DISCUSSION THREAD

**Discussion by hebeillc** [tokio-rs/axum#2413] [Q&A] [answered]
### Summary

I created a route as explained in [.](https://docs.rs/axum/latest/axum/struct.Router.html#captures) When requesting a route, passing Chinese will not be executed, while passing English will be normal. What should I do?
My code is:
`
use std::error::Error;
use tokio::{net::TcpListener};
use axum::{Router, routing::{get, Route, delete, post}, extract::Path, http::uri::PathAndQuery};
#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>>{
    let app = get_routes().await;
    let tcp = TcpListener::bind("127.0.0.1:8080").await?;
    axum::serve(tcp, app).await?;
    Ok(())
}
async fn get_routes() -> Router {
    let app = Router::new().route("/user/:name", get(get_user));
    app
}
async fn get_user(Path(name):Path<String>) -> String{
    println!("name={}", name);
    format!("name is {name}")
}
`
I request “127.0.0.1:8080/user/tom” It can run, but if i request "127.0.0.1:8080/user/李", It won't run

### axum version

0.7.2

**Comment by davidpdrsn** [comment, marked as answer]
Thats most likely a bug in [matchit](https://github.com/ibraheemdev/matchit) which is the routing library axum uses internally.

**Reply by hebeillc** [reply]
Okay, thank you. I'll go to [matchit](https://github.com/ibraheemdev/matchit) and ask a question

**Comment by ttys3** [comment]
the route works fine. I do not think axum or [matchit](https://github.com/ibraheemdev/matchit)  has issue.

you should check you code.

and next time, please DO format you code before you asking someone.

**Reply by hebeillc** [reply]
Sorry, my problem. I tested it with a browser and there was no problem. It was an issue with the API platform I used
